### PR TITLE
Create Ascii_subsequences.java

### PR DESCRIPTION
**Pull request by Bhavyaa08**
Ascii Subsequences Problem java solution and IsBST java solution

Issue No : #1677 and #1678

Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Languages Used:
- Java

#### Files Added:
- Ascii_Subsequences.java
- isBST.java

### CODE REVIEW

File: ASCII_SUBSEQUENCES/Ascii_subsequences.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import java.util.Scanner;
+
+public class Ascii_Subsequences {
+	public static void main(String args[]) {
```
Comment (SakshiShreya) [inline ASCII_SUBSEQUENCES/Ascii_subsequences.java]: Use 4 spaces instead of 1 tab

File: Is_Binary_Search_Tree/isBST.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	private boolean isBST(Node node) {
+
+		if (node == null)
+			return true;
+
+		boolean l = isBST(node.left); // l will store whether the left subtree is BST or not
+		boolean r = isBST(node.right); // r will store whether the right subtree is BST or not
+
+		//the below given conditions will return false if the current node doesnt satisfy the property of a BST
+		if(node.left!=null && node.data < node.left.data)
```
Comment (SakshiShreya) [inline Is_Binary_Search_Tree/isBST.java]: Add a space around operators.

File: Is_Binary_Search_Tree/isBST.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import java.util.Scanner;
+
+public class BinaryTree {
+
+	class Node {
+
+		int data;
+		Node left;
+		Node right;
+
```
Comment (SakshiShreya) [inline Is_Binary_Search_Tree/isBST.java]: Please remove these extra new lines.

File: Is_Binary_Search_Tree/isBST.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+			return true;
+
+		boolean l = isBST(node.left); // l will store whether the left subtree is BST or not
+		boolean r = isBST(node.right); // r will store whether the right subtree is BST or not
+
+		//the below given conditions will return false if the current node doesnt satisfy the property of a BST
+		if(node.left!=null && node.data < node.left.data)
+			return false;
+
+		if(node.right!=null && node.data>node.right.data)
```
Comment (SakshiShreya) [inline Is_Binary_Search_Tree/isBST.java]: Add a space around operators.

File: Is_Binary_Search_Tree/isBST.java [hunk 5]
```diff
@@ -1,18 +1,14 @@
 import java.util.Scanner;
 
 public class BinaryTree {
-
 	class Node {
```
Comment (SakshiShreya) [inline Is_Binary_Search_Tree/isBST.java]: There are still tabs in this file.

**Comment by SakshiShreya** [comment]
Also mention the issue numbers after Fixes #.
And work on only one issue per PR.

**Comment by Bhavyaa08** [comment]
I have made a new pull request as I talked to you about previously.

**Pull request closed without merging**